library/block/block_content_registry/src/main/java/org/quiltmc/qsl/block/content/registry/impl/BlockContentRegistriesInitializer.java

```diff
@@ -0,0 +1,126 @@
+ * limitations under the License.
+ */
+
+package org.quiltmc.qsl.block.content.registry.impl;
+
+import java.util.Map;
+import java.util.function.BiFunction;
+import java.util.stream.Collectors;
+
+import com.google.common.collect.BiMap;
+import com.google.common.collect.HashBiMap;
+import com.google.common.collect.ImmutableMap;
+import org.quiltmc.loader.api.ModContainer;
+import org.quiltmc.qsl.base.api.entrypoint.ModInitializer;
+import org.quiltmc.qsl.block.content.registry.api.BlockContentRegistries;
+import org.quiltmc.qsl.block.content.registry.api.FlammableBlockEntry;
+import org.quiltmc.qsl.block.content.registry.api.ReversibleBlockEntry;
+import org.quiltmc.qsl.registry.attachment.api.RegistryEntryAttachment;
+import org.quiltmc.qsl.resource.loader.api.ResourceLoaderEvents;
+
+import net.minecraft.block.Block;
+import net.minecraft.block.BlockState;
+import net.minecraft.block.Blocks;
+import net.minecraft.block.FireBlock;
+import net.minecraft.block.Oxidizable;
+import net.minecraft.block.SculkSensorBlock;
+import net.minecraft.item.AxeItem;
+import net.minecraft.item.HoneycombItem;
+import net.minecraft.item.ShovelItem;
+import net.minecraft.world.event.GameEvent;
+
+public class BlockContentRegistriesInitializer implements ModInitializer {
+	private static final Map<Block, BlockState> INITIAL_PATH_STATES = ImmutableMap.copyOf(ShovelItem.PATH_STATES);
+	private static final Map<Block, Block> INITIAL_STRIPPED_BLOCKS = ImmutableMap.copyOf(AxeItem.STRIPPED_BLOCKS);
+
+	public static final BiMap<Block, Block> INITIAL_OXIDATION_BLOCKS = HashBiMap.create();
+	public static final BiMap<Block, Block> OXIDATION_INCREASE_BLOCKS = HashBiMap.create();
+	public static final BiMap<Block, Block> OXIDATION_DECREASE_BLOCKS = HashBiMap.create();
+
+	public static final BiMap<Block, Block> INITIAL_WAXED_BLOCKS = HashBiMap.create();
+	public static final BiMap<Block, Block> WAXED_UNWAXED_BLOCKS = HashBiMap.create();
+	public static final BiMap<Block, Block> UNWAXED_WAXED_BLOCKS = HashBiMap.create();
+
+	private static final Map<Block, FlammableBlockEntry> INITIAL_FLAMMABLE_BLOCKS;
+
+	static {
+		var builder = ImmutableMap.<Block, FlammableBlockEntry>builder();
+		FireBlock fireBlock = ((FireBlock) Blocks.FIRE);
+		fireBlock.spreadChances.keySet().forEach(block -> builder.put(block, new FlammableBlockEntry(fireBlock.burnChances.getInt(block), fireBlock.spreadChances.getInt(block))));
+		INITIAL_FLAMMABLE_BLOCKS = builder.build();
+	}
+
+	private static final Map<GameEvent, Integer> INITIAL_SCULK_SENSOR_BLOCK_EVENTS = ImmutableMap.copyOf(SculkSensorBlock.EVENTS);
+
+	@Override
+	public void onInitialize(ModContainer mod) {
+		// Force load the maps
+		Oxidizable.OXIDATION_LEVEL_INCREASES.get();
+		HoneycombItem.UNWAXED_TO_WAXED_BLOCKS.get();
+
+		resetMaps();
+		ResourceLoaderEvents.END_DATA_PACK_RELOAD.register((server, resourceManager, error) -> resetMaps());
+	}
+
+	private static void resetMaps() {
```

Comment:
This approach should be replaced with ICR's new take on it, so the vanilla entries can be modified in-code